doc/Data Plane Code Generation/Subclients.md

```diff
@@ -0,0 +1,247 @@
+# Customizing the generated code
+
+## Subclients
+
+There are two categories of clients: service clients and their subclients. Service clients can be instantiated and have the Client suffix. Subclients can only be created by calling factory methods on other clients (commonly on service clients) and do not have the client suffix. The service client is the entry point to the API for an Azure service. Please refer to [guideline](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-subclients) for more detail information.
+
+There are two tools to organize subclients and service client.
+
+- **Single top-level client**: define one top-level service client with the name <service_name>Client, all other clients will be its subclient.
+
+- **client hierarchy**: organize the client service and subclients via defining the parent-subclient hierarchy dependency.
+
+**Note**: You can combine those two approaches to define an client hierarchy with one top-level client and its subclient may have sub clients also. 
+
+### Single top-level client
+
+With multiple service clients it may be better to group them all under one top-level client.
+
+<details>
+
+**Generated code before:**
+
+``` C#
+//Generated\RootClient.cs
+namespace Azure.Service.SubClients
+{
+    public partial class RootClient
+    {
+        public RootClient(string cachedParameter, AzureKeyCredential credential, Uri endpoint = null, SubClientsClientOptions options = null){}
+    }
+}
+
+//Generated\ParameterClient.cs
+namespace Azure.Service.SubClients
+{
+    public partial class ParameterClient
```

Comment:
I think it would be more helpful to our service team partners if we gave an example of clients that are named more like service functionality.  "ParameterClient" is the name of a client that tests parameters are generated, not a client that calls an Azure service.

Reply:
I updated to Client2Client